Use generic http.Client

adamreese opened the pull request:
This change allows for overriding the http.Client. It will enable using other wasi http SDK's.

...and some random formatting

radu-matei reviewed (approved):
🎉

rajatjindal commented on client.go:
```diff
@@ -35,6 +33,9 @@ type Config struct {
 // Client provides an interface for interacting with the S3 API.
 type Client struct {
+	// HTTPClient overrides the default http.Client.
+	HTTPClient *http.Client
```
we can keep this client private. do you see any reason for keeping this public?

adamreese replied:
I went back and forth on where to include it. Most cases it will not be used so I removed it from the `New(...)` function parameters. It didn't feel right in `Config{}` either but that might be the correct place. Functional options felt heavy. I was hoping to get your thoughts on it.

rajatjindal replied:
although wasihttp modifies the http.DefaultClient to use transport provided by wasihttp sdk, I would like to keep this as an explicit option. 

maybe we can have a WithHTTPClient function like a lot of other sdk provides?

adamreese replied:
Updated with your suggestion

adamreese committed:
Use generic http.Client

This change allows for overriding the http.Client. It will enable using
other wasi http SDK's.

Signed-off-by: Adam Reese <adam@reese.io>
